fendermint/app/src/app.rs

```diff
@@ -76,12 +76,33 @@ impl AppState {
     pub fn state_root(&self) -> Cid {
         self.state_params.state_root
     }
+
     pub fn chain_id(&self) -> ChainID {
         ChainID::from(self.state_params.chain_id)
     }
+
+    /// Produce an appliction hash that is a commitment to all data replicated by consensus,
+    /// that is, all nodes participating in the network must agree on this otherwise we have
+    /// a consensus failure.
+    ///
+    /// Notably it contains the actor state root _as well as_ some of the metadata maintained
+    /// outside the FVM, such as the timestamp and the circulating supply.
     pub fn app_hash(&self) -> tendermint::hash::AppHash {
-        tendermint::hash::AppHash::try_from(self.state_root().to_bytes())
-            .expect("hash can be wrapped")
+        // Create an artifical CID from the FVM state params, which include everything that
+        // deterministically changes under consensus.
+        let state_params_cid =
+            fendermint_vm_message::cid(&self.state_params).expect("state params have a CID");
+
+        // We could reduce it to a hash to ephasize that this is not something that we can return at the moment,
```

Comment:
Yes, the `state_root_hash` indeed comes from the `state_param`: https://github.com/consensus-shipyard/fendermint/blob/4efd23d568bfd440804879b0fa0f471b6f6fe146/fendermint/app/src/app.rs#L77

The reason we can't retrieve it is because we store it as part of the app state history: https://github.com/consensus-shipyard/fendermint/blob/4efd23d568bfd440804879b0fa0f471b6f6fe146/fendermint/app/src/app.rs#L266-L278 which doesn't use the `Blockstore` but a separate collection in the KV store.